gcloud-java-compute/src/main/java/com/google/gcloud/compute/AddressInfo.java

```diff
@@ -0,0 +1,572 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.compute.model.Address;
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.io.Serializable;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Google Compute Engine address. With Compute Engine you can create static external IP addresses
+ * that are assigned to your project and persists until you explicitly release them. A region
+ * address can be assigned to a Compute Engine instance or to a regional forwarding rule. To create
+ * a region address use a {@link RegionAddressId} identity. Compute Engine also allows to create
+ * global addresses that are used for global forwarding rules. Both global addresses and global
+ * forwarding rules can only be used for HTTP load balancing. To create a global address use a
+ * {@link GlobalAddressId} identity.
+ *
+ * @see <a href="https://cloud.google.com/compute/docs/instances-and-network#reservedaddress">
+ *     Static external IP addresses</a>
+ * @see <a href="https://cloud.google.com/compute/docs/load-balancing/http/">HTTP Load Balancing</a>
+ */
+public class AddressInfo implements Serializable {
+
+  static final Function<Address, AddressInfo> FROM_PB_FUNCTION =
+      new Function<Address, AddressInfo>() {
+        @Override
+        public AddressInfo apply(Address pb) {
+          return AddressInfo.fromPb(pb);
+        }
+      };
+  static final Function<AddressInfo, Address> TO_PB_FUNCTION =
+      new Function<AddressInfo, Address>() {
+        @Override
+        public Address apply(AddressInfo addressInfo) {
+          return addressInfo.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = 7678434703520207500L;
+  private static final DateTimeFormatter TIMESTAMP_FORMATTER = ISODateTimeFormat.dateTime();
+
+  private final String address;
+  private final Long creationTimestamp;
+  private final String description;
+  private final String id;
+  private final AddressId addressId;
+  private final Status status;
+  private final Usage usage;
+
+  /**
+   * The status of the address.
+   */
+  public enum Status {
+
+    /**
+     * The address is reserved for the project and is available for use.
+     */
+    RESERVED,
+
+    /**
+     * The address is currently being used and thus not available.
+     */
+    IN_USE
+  }
+
+  /**
+   * Base class for a Google Compute Engine address usage information. {@link InstanceUsage} is for
+   * addresses assigned to a Google Compute Engine instance.
```

Comment:
I do not really understand this comment. Can you rephrase?

{@link InstanceUsage} is for addresses assigned to a Google Compute Engine instance.